test/functional/test-amp-img.js

```diff
@@ -374,4 +376,93 @@ describe('amp-img', () => {
     impl.unlayoutCallback();
   });
 
+  describe('blurred image placeholder', () => {
+    let el;
+    let img;
+    let impl;
+
+    beforeEach(() => {
+      toggleExperiment(iframe.win, 'blurry-placeholder', true);
+      getImgWithBlur(true, true);
+    });
+
+    afterEach(() => {
+      impl.unlayoutCallback();
+    });
+
+    function getImgWithBlur(addPlacholder, addBlurClass) {
+      el = document.createElement('amp-img');
+      img = document.createElement('img');
+      if (addPlacholder) {
+        img.setAttribute('placeholder', '');
+      }
+      if (addBlurClass) {
+        img.classList.add('i-amphtml-blur');
+      }
+      el.appendChild(img);
+      el.getResources = () => Services.resourcesForDoc(document);
+      el.getPlaceholder = () => {return img;};
+      impl = new AmpImg(el);
+    }
+
+    it('should correctly detect if a blurry image child exists', () => {
+      impl.buildCallback();
+      impl.layoutCallback();
+      expect(impl.hasBlurredPlaceHolder_).to.be.true;
+
+      getImgWithBlur(true, false);
+      impl.buildCallback();
+      impl.layoutCallback();
+      expect(impl.hasBlurredPlaceHolder_).to.be.false;
+
+      el = document.createElement('amp-img');
+      el.getPlaceholder = () => {return undefined;};
+      impl = new AmpImg(el);
+      impl.buildCallback();
+      impl.layoutCallback();
+      expect(impl.hasBlurredPlaceHolder_).to.be.false;
+
+      el = document.createElement('amp-img');
+      el.getPlaceholder = () => {return undefined;};
+      impl = new AmpImg(el);
+      impl.buildCallback();
+      impl.layoutCallback();
+      expect(impl.hasBlurredPlaceHolder_).to.be.false;
+    });
+
+    it('should fade in the image if the blurry placheolder exists', () => {
+      impl.buildCallback();
+      impl.layoutCallback();
+      expect(impl.element.classList.contains(
+          'i-amphtml-blur-loading')).to.be.true;
```

Comment:
Sorry if these expect statements are hard to read -I had to break them up to satisfy the lint rule of being <80 chars on a line. If there is a better place for me to break it/an alternate solution, then please let me know!